### ISSUE TITLE
List users endpoint with filtering, pagination

### ISSUE THREAD

**Issue by dallashuggins** [deptagency/algomart#242]

**Comment by lefnire** [comment]
@dallashuggins for adding a user, how do we want to handle stuff that'd normally be input by the user? Eg password & passphrase? I'm thinking password could be the typical temporary->user-update workflow, but is passphrase a 1-time set?

**Comment by dallashuggins** [comment]
@lefnire Aw right, great point. I totally forgot about this aspect, and remember now why pagination and filtering was important. It may actually make sense to search through users by email, and then assign/revoke permissions, vs creating the user where we they would have to share that sensitive information. The passphrase is only set once and cannot be reset yet, so that works much better. Sorry for the misleading comments! Remembering there is one complication there too, as Firebase Admin doesn't allow for easy filtering through the list of users, so we'll need to search based on the exact email or phone number.

**Comment by lefnire** [comment]
To verify: no create-user feature here then (to avoid admin needing passphrase)? 

> as Firebase Admin doesn't allow for easy filtering through the list of users, so we'll need to search based on the exact email or phone number.

Username & email are stored in api/UserAccount as well, we could `ILIKE`. Not phone though, do we indeed want the search against Firebase users, not UserAccounts?

**Comment by dallashuggins** [comment]
@lefnire Yeah, I think that makes the most sense. Oh great call! I hadn't thought about sorting through the users table actually, but there should always be a Firebase account associated, so that would work perfectly. 🙌
